rasa/nlu/components.py

```diff
@@ -322,6 +341,70 @@ def validate_required_components_from_data(
         )
 
 
+def warn_of_competing_extractors(pipeline: List["Component"]) -> None:
+    """Warns the user when using competing extractors.
+
+    Competing extractors are e.g. `CRFEntityExtractor` and `DIETClassifier`.
+    Both of these look for the same entities based on the same training data
+    leading to ambiguity in the results.
+    """
+    general_purpose_entity_extractors = [
+        "DIETClassifier",
+        "CRFEntityExtractor",
+        "MitieEntityExtractor",
+    ]
+    extractors_in_pipeline = find_components_in_pipeline(
+        general_purpose_entity_extractors, pipeline
+    )
+    if len(extractors_in_pipeline) > 1:
+        rasa.shared.utils.io.raise_warning(
+            f"You have defined multiple entity extractors that do the same job "
+            f"in your pipeline: "
+            f"{', '.join(extractors_in_pipeline)}. This can lead to the same entity getting "
+            f"extracted multiple times. Please read the documentation section "
+            f"on entity extractors to make sure you understand the implications: "
+            f"{DOCS_URL_COMPONENTS}#entity-extractors"
+        )
+
+
+def warn_of_competition_with_regex_extractor(
+    pipeline: List["Component"], data: TrainingData
+) -> None:
+    """Warns when regex entity extractor is competing with a general one.
+
+    This might be the case when the following conditions are all met:
+    * You are using a general entity extractor and the `RegexEntityExtractor`
+    * AND you have regex patterns for entity type A
+    * AND you have annotated text examples for entity type A
+    """
+    general_entity_extractors = [
+        "DIETClassifier",
```

Comment:
can we use constants for these? E.g. like [here](https://github.com/RasaHQ/rasa/blob/8258c5500108110525aef16fa20ee0f47f435b1e/rasa/shared/core/constants.py#L89)?

Reply:
Would you be okay with creating an extra issue for this as a potential improvement? The names as strings are used throughout the components file. I think this is because the components themselves derive their name property directly from their class name. If you create constants you also need to have another mechanism that makes sure the constants stay in sync with the class names. So it's a relevant change, but a bit more involved than just a side note inside this PR

Reply:
Also the component names are used in many many more places. Like when creating pipelines for test code

Reply:
Sure - could you please create an issue for this then?

Reply:
yup: https://github.com/RasaHQ/rasa/issues/8455